docs/triggertemplates.md

```diff
@@ -0,0 +1,52 @@
+# TriggerTemplates
+A `TriggerTemplate` is a resource that can template resources. `TriggerTemplates` have optional parameters that can be substituted anywhere within the resource template. If resources do not have a name specified, it will default to value of the resource kind. Further, all resources names will have a unique timestamp postfix to avoid naming conflicts.
+
+```YAML
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: simple-pipeline-template
+  namespace: default
+spec:
+  params:
+    - name: gitrevision
+      description: The git revision
+      default: master
+    - name: gitrepositoryurl
+      description: The git repository url
+    - name: namespace
+      description: The namespace to create the resources
+  resourcetemplates:
+    - apiVersion: tekton.dev/v1alpha1
+      kind: PipelineResource
+      metadata:
+        name: git-source
+        namespace: ${params.namespace}
+        labels:
+            triggertemplated: true
+      spec:
+        type: git
+        params:
+        - name: revision
+          value: ${params.gitrevision}
+        - name: url
+          value: ${params.namespace}
+    - apiVersion: tekton.dev/v1alpha1
+      kind: PipelineRun
+      metadata:
+        name: simple-pipeline-run
+        namespace: default
+        labels:
+            triggertemplated: true
+      spec:
+        pipelineRef:
+            name: simple-pipeline
+        trigger:
+          type: event
+        resources:
+        - name: git-source
+          resourceRef:
+            name: git-source
+```
+
+Similar to [Pipelines](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md), `TriggerTemplates` do not do any actual work, but instead act as the blueprint for what resources should be created. Parameters defined in a `TriggerTemplate` are overridden by those of the same name in a [`TriggerBinding`](triggerbindings.md).
```

Comment:
2 total nitpicks:
- I think we usually wrap long lines like this in md files
- "are overridden by" confuses me a little, maybe something like "can be provided by"? (same comment applies to the TriggerBindings docs also)

Reply:
I modified the files so each sentence is a new line. That seems like the easiest way to wrap things more or less?